tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -508,36 +510,48 @@ private[emitter] final class ScalaJSClassEmitter(semantics: Semantics,
   }
 
   /** Generate `classVar.prototype.name = value` */
-  def genAddToPrototype(className: String, name: js.PropertyName,
-      value: js.Tree)(implicit pos: Position): js.Tree = {
+  def genAddToPrototype(className: String, name: js.PropertyName, value: js.Tree)(
+      implicit globalKnowledge: GlobalKnowledge, pos: Position): js.Tree = {
     genAddToObject(encodeClassVar(className).prototype, name, value)
   }
 
   /** Generate `classVar.prototype.name = value` */
-  def genAddToPrototype(className: String, name: PropertyName,
-      value: js.Tree)(implicit pos: Position): js.Tree = {
+  def genAddToPrototype(className: String, name: PropertyName, value: js.Tree)(
+      implicit globalKnowledge: GlobalKnowledge, pos: Position): js.Tree = {
     genAddToPrototype(className, genPropertyName(name), value)
   }
 
   /** Generate `obj.name = value` */
   def genAddToObject(obj: js.Tree, name: js.PropertyName,
       value: js.Tree)(implicit pos: Position): js.Tree = {
-    val select = name match {
-      case name: js.Ident         => js.DotSelect(obj, name)
-      case name: js.StringLiteral => genBracketSelect(obj, name)
-    }
-    js.Assign(select, value)
+    js.Assign(genPropSelect(obj, name), value)
   }
 
   /** Generate `obj.name = value` */
-  def genAddToObject(obj: js.Tree, name: PropertyName,
-      value: js.Tree)(implicit pos: Position): js.Tree = {
+  def genAddToObject(className: String, obj: js.Tree, name: PropertyName,
+      value: js.Tree)(
+      implicit globalKnowledge: GlobalKnowledge, pos: Position): js.Tree = {
```

Comment:
It seems like this would make more sense:

```scala
def genAddToObject(className: String, obj: js.Tree, name: PropertyName,
    value: js.Tree)(implicit globalKnowledge: GlobalKnowledge,
    pos: Position): js.Tree =
```

Though looking at it, it looks weird as well :P and there is definitely value in having the implicit on the new line :)